### PR TITLE
[Perf] [Fun-CosyVoice3] AR incremental streaming output

### PR DESCRIPTION
**Pull request by Kozmosa**
## Motivation

Part of #1652 (AR: Streaming output).

The AR-side producer for incremental streaming: emit Fun-CosyVoice3 speech tokens as they are generated, following the MOSS-TTS / Qwen3-TTS `stream_output_builder` pattern, so a streaming vocoder can start decoding before generation finishes.

## Modifications

1. `make_cosyvoice3_stream_output_builder()` ships the not-yet-emitted tail of `output_codes` after every AR step. Control ids are dropped at collection, so a per-request cursor gives exactly-once delivery. Emission is suppressed while chunked prefill is in flight.

2. Chunks are 1-D long CPU tensors to the vocoder stage, with `modality=audio_codes`, `sample_rate=24000`, and a `row_index` cursor in the metadata.

3. The stream gate reads the prepared state dict first and falls back to request params for unprepared payloads.

The builder is deliberately not wired into the scheduler in this PR. With the buffered vocoder there is no safe intermediate state: the stage runtime rejects stream chunks that arrive before the request payload unless the receiving stage opted in, and the buffered SimpleScheduler cannot consume them either. Activation belongs with the DiT chunk-wise vocoder, which lands the `stream_to` wiring, the receiver flag and the consumer together. #1656 implements that full path; this PR then serves as the producer-side reference for the scheduler-builder pattern.

## Related Issues

Part of #1652.

## Accuracy Test

- `tests/unit_test/fun_cosyvoice3/`: 47 passed
- Full `tests/unit_test` suite and pre-commit pass
- During development the builder was wired and smoke-tested end to end on H100 with the real checkpoint; that run is what surfaced the pre-payload rejection described above

## Checklist

- [x] Format your code according with pre-commit.
- [x] Add unit tests.
- [ ] Update documentation / docstrings / example tutorials as needed.
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed.
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.

## CI

CI runs on self-hosted GPU runners and requires a maintainer to add the
`run-ci` label. Once labeled, every subsequent push re-triggers CI as
long as the label remains. Use `/tag-and-rerun-ci higgs` or
`/tag-and-rerun-ci moss` to select a TTS CI model, and
`/tag-and-rerun-ci fun-asr`, `/tag-and-rerun-ci qwen3-asr` or
`/tag-and-rerun-ci whisper-asr` to select an ASR CI model. One selector from each family can be combined, for example
`/tag-and-rerun-ci moss fun-asr`. Draft PRs are skipped even if labeled.

### CODE REVIEW

File: sglang_omni/models/fun_cosyvoice3/request_builders.py [hunk 1]
```diff
@@ -840,3 +844,69 @@ def result_adapter(data: CosyVoice3SGLangRequestData) -> StagePayload:
...
+
+def _cosyvoice3_stream_enabled(payload: StagePayload | None) -> bool:
+    """Streaming gate: the prepared state dict is authoritative (it folds the
+    ``tts_params``/``params`` stream flags parsed by ``build_cosyvoice3_state``);
+    the request params are only a fallback for unprepared payloads."""
+    data = getattr(payload, "data", None)
+    if isinstance(data, dict) and bool(data.get("stream", False)):
+        return True
+    params = getattr(getattr(payload, "request", None), "params", None)
+    return isinstance(params, dict) and bool(params.get("stream", False))
```
[the remark this answers is not in the document]
Comment (Kozmosa) [reply sglang_omni/models/fun_cosyvoice3/request_builders.py:857]: fixed in ad9051a. the gate now honors the state dict whenever the stream key is there, so an explicit false wins over params. btw nothing writes stream into tts_params today, so this was latent, not reachable

File: tests/unit_test/fun_cosyvoice3/test_stream_output.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+def test_stream_builder_falls_back_to_request_params_for_unprepared_payloads() -> None:
+    builder = make_cosyvoice3_stream_output_builder()
+    payload = _payload(data={}, params={"stream": True})
+    data = _data(payload, codes=[7])
+
+    messages = builder("req-cosy", data, None)
+    assert len(messages) == 1
+    assert messages[0].data.tolist() == [7]
+
```
[the remark this answers is not in the document]
Comment (Kozmosa) [reply tests/unit_test/fun_cosyvoice3/test_stream_output.py:119]: done, test_stream_builder_state_dict_stream_false_overrides_params covers it. ad9051a

**Comment by Kozmosa** [comment]
reworked this after looking harder at the wiring question. an earlier revision wired the builder behind a supports_streaming_vocoder gate, but that was me inventing a proxy for what the framework already expresses: streaming needs stream_to, the receiver flag and a stream-aware consumer scheduler to land together, there is no accept-and-drop middle ground. so the gate is gone and the builder ships unwired, activation happens with the diT chunk-wise vocoder. #1656 already implements the full path, so this is mostly a reference implementation of the producer side now

File: tests/unit_test/fun_cosyvoice3/test_stream_output.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    payload: StagePayload,
+    *,
+    codes: list[int],
+    req: Any = SimpleNamespace(inflight_middle_chunks=0),
+) -> CosyVoice3SGLangRequestData:
+    return CosyVoice3SGLangRequestData(
+        req=req,
+        stage_payload=payload,
+        output_codes=[torch.tensor([code], dtype=torch.long) for code in codes],
+    )
```
[the remark this answers is not in the document]
Comment (Kozmosa) [reply tests/unit_test/fun_cosyvoice3/test_stream_output.py:45]: good point, switched the default to a sentinel and build a fresh namespace per call. none of the current tests mutate the default req so nothing was failing, but the footgun was real. 71314af

**Comment by PoTaTo-Mika** [comment]
#1656 You may need to corporate with zhihao.

**Comment by Kozmosa** [comment]
reaching out to @guozhihao-224 on #1656 now. happy to close this in favor of #1656 once the ar side is settled, keeping it open for now as the producer-pattern reference mentioned in the description
